lib/SPIRV/OCLUtil.cpp

```diff
@@ -1292,6 +1293,8 @@ class OCLBuiltinFuncMangleInfo : public SPIRV::BuiltinFuncMangleInfo {
   // Auxiliarry information, it is expected that it is relevant at the moment
   // the init method is called.
   Function *F;                  // SPIRV decorated function
+  // TODO: ArgTypes argument should get removed once all SPV-IR related issues
+  // are resolved
```

Comment:
`OCLBuiltinFuncMangleInfo` should depend only on `SPV-IR` function.

As you can see here: https://github.com/KhronosGroup/SPIRV-LLVM-Translator/blob/master/lib/SPIRV/OCLUtil.cpp#L1295 `ArgTypes` is a vector that keeps arguments of **OpenCL** builtin - these arguments were created by logic which is OCL-specific and should be placed in SPIRVToOCL passes instead of SPIRVReader. We still have some builtins that are translated into OCL-IR in SPIRVReader and these builtins use `ArgTypes` vector. Once we move all the OCL-specific logic into SPIRVToOCL passes, we could depend only on `getArgTy` function which returns SPV-IR arguments types.

Reply:
Understood, thank you.